Source/EasyNetQ.Management.Client/UnexpectedHttpStatusCodeException.cs

```diff
@@ -48,4 +35,47 @@ StreamingContext context
     ) : base(info, context)
     {
     }
+
+    protected UnexpectedHttpStatusCodeException(string message, HttpStatusCode statusCode) : base(message)
+    {
+        StatusCode = statusCode;
+    }
+
+    public static async Task<UnexpectedHttpStatusCodeException> FromHttpResponseMessageAsync(HttpResponseMessage response)
```

Comment:
You also might consider adding cancellation token here...

Reply:
@Pliner well, ok.

However, [HttpContent.ReadAsStringAsync(CancellationToken)](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpcontent.readasstringasync?view=net-8.0#system-net-http-httpcontent-readasstringasync(system-threading-cancellationtoken)) exists since .NET 5.0 only.